Adds ENTERPRISE_PUBLIC_ENROLLMENT_API_URL to settings

pomegranited opened the pull request:
cf https://github.com/edx/edx-enterprise/pull/67

* `ENTERPRISE_PUBLIC_ENROLLMENT_API_URL`'s hostname uses the ENV-configured, fully-qualified `LMS_ROOT_URL`, so it can be used when making external, client-side requests to the Enrollment API.
* `ENTERPRISE_ENROLLMENT_API_URL`'s hostname uses `localhost:8000`, for use when making internal, server-side requests to the Enrollment API.

**Sandbox URL**:

* LMS: https://pr14671.sandbox.opencraft.hosting/
* Studio: https://studio-pr14671.sandbox.opencraft.hosting/

Running 036715c

**Deployment targets**: edx.org and edge.edx.org

**Testing Instructions:**

See https://github.com/edx/edx-enterprise/pull/67

**Reviewers**:

- [x] @haikuginger 
- [x] @mattdrayer

edx-webhook commented:
Thanks for the pull request, @pomegranited! It looks like you're a member of a company that does contract work for edX. If you're doing this work as part of a paid contract with edX, you should talk to edX about who will review this pull request. If this work is not part of a paid contract with edX, then you should ensure that there is an OSPR issue to track this work in [JIRA](https://openedx.atlassian.net), so that we don't lose track of your pull request.

<a   href="https://openedx-webhooks.herokuapp.com/github/process_pr?repo=edx%2Fedx-platform&number=14671"   rel="nofollow" >Create an OSPR issue for this pull request</a>.

pomegranited commented:
jenkins run js

haikuginger commented:
@mattdrayer @brittneyexline, can you sign off on this, please?

mattdrayer commented:
Any way to add a test to cover this new setting?

pomegranited commented:
@haikuginger I've addressed your comment with [0f8ef7b
](https://github.com/edx/edx-platform/pull/14671/commits/0f8ef7b4390f4aecabf661b63f9a048aeff7d9d9#diff-8a23909bc8d201fba8525b1dd6396247R915), and added comments explaining each of the settings.

pomegranited commented:
@mattdrayer Re tests, please see https://github.com/edx/edx-enterprise/pull/67#issuecomment-287215412

mattdrayer commented:
jenkins run js

pomegranited commented:
@mattdrayer I'm going to rebase to see if that fixes the JS tests.

pomegranited commented:
Rebased 0f8ef7b

mattdrayer commented:
👍 assuming everything turns green ☘️ ☘️ ☘️

pomegranited commented:
@haikuginger @mattdrayer 

I'm going to have to do an edx-enterprise version bump after https://github.com/edx/edx-enterprise/pull/67 merges to enact this change.. Should I do that here, or create a separate edx-platform PR?

mattdrayer commented:
We can handle it either way, although if you include the version upgrade in this PR then we have explicit context about the need for the change.

pomegranited commented:
@mattdrayer  Ok cool, I'm going to merge https://github.com/edx/edx-enterprise/pull/67 and include the version bump in this PR.

pomegranited commented:
@mattdrayer @haikuginger Version bumped, and I've updated the sandbox.

Could I please get another round of 👍 s?  (Or :shamrock: if you prefer 😄 )

mattdrayer commented:
:shamrock: :shamrock: :shamrock:

haikuginger commented:
👍
